ast_releasenotes/releasenotes/notes/extended-gate-syntax-15b2f58b731db7e4.yaml

```diff
@@ -0,0 +1,8 @@
+---
+features:
+  - |
+    Introduces a new ast node type ``QuantumExtendedGateDefinition``` where the arguments
+    are ``List[ClassicalType]`` instead of ``List[Indentifier]``. A new node type was
+    introduced in order to avoid breaking backward compatibility with prior versions of
```

Comment:
I think even if we were OK with breaking backwards compatibility, there is still the question of how to distinguish the two kinds of gate definitions. The only alternative that comes to mind is to have some extra property on `QuantumGate` to flag that it is an extended definition (and this isn't any better than just adding a new definition, IMO).

Reply:
To echo something I mentioned in the TSC meeting, as an implementor I prefer the `QuantumExtendedGateDefinition` since adding a new type like this gives a very legible "blast radius" to any necessary code changes I'd need to support (namely, additive code changes to some visitors).